sw/device/lib/dif/dif_spi_host.h

```diff
@@ -4,20 +4,193 @@
 
 #ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_SPI_HOST_H_
 #define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_SPI_HOST_H_
-
 /**
  * @file
- * @brief <a href="/hw/ip/spi_host/doc/">SPI Host</a> Device Interface
- * Functions
+ * @brief <a href="/hw/ip/spi_host/doc/">SPI Host</a> Device Interface Functions
  */
 
+#include <stdint.h>
+
 #include "sw/device/lib/dif/autogen/dif_spi_host_autogen.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif  // __cplusplus
 
-// TODO: Add DIFs.
+/**
+ * Runtime configuration for SPI Host.
+ *
+ * This struct describes (SOFTWARE) runtime information for one-time
+ * configuration of the hardware.
+ */
+typedef struct dif_spi_host_config {
+  uint32_t spi_clock;
+  uint32_t peripheral_clock_freq_hz;
+  struct {
+    uint8_t idle;
+    uint8_t trail;
+    uint8_t lead;
+  } chip_select;
+  bool full_cycle;
+  bool cpha;
+  bool cpol;
+} dif_spi_host_config_t;
+
+/**
+ * Width of SPI operations.
+ */
+typedef enum dif_spi_host_width {
+  /** Standard SPI mode (single lanes for send and recv). */
+  kDifSpiHostWidthStandard = 0,
+  /** Dual SPI mode (use two lanes for send and recv). */
+  kDifSpiHostWidthDual = 1,
+  /** Quad SPI mode (use four lanes for send and recv). */
+  kDifSpiHostWidthQuad = 2,
+} dif_spi_host_width_t;
+
+/**
+ * Direction of SPI operations.
+ *
+ * This describes which direction a given SPI operation will use.
+ */
+typedef enum dif_spi_host_direction {
+  /** The SPI host neither transmits or receives. */
+  kDifSpiHostDirectionDummy = 0,
+  /** The SPI host receives data. */
+  kDifSpiHostDirectionRx = 1,
+  /** The SPI host transmits data. */
+  kDifSpiHostDirectionTx = 2,
+  /** The SPI host transmits and receives data. */
+  kDifSpiHostDirectionBidirectional = 3,
+} dif_spi_host_direction_t;
+
+/**
+ * Segment types for segments in a transaction.
+ *
+ */
+typedef enum dif_spi_host_segment_type {
+  /** The segment is a SPI opcode */
+  kDifSpiHostSegmentTypeOpcode,
+  /** The segment is a SPI address */
+  kDifSpiHostSegmentTypeAddress,
+  /** The segment is a SPI dummy cycle */
+  kDifSpiHostSegmentTypeDummy,
+  /** The segment is a SPI transmit (from a memory buffer) */
+  kDifSpiHostSegmentTypeTx,
+  /** The segment is a SPI receive (into a memory buffer) */
+  kDifSpiHostSegmentTypeRx,
+  /** The segment is a simultaneous transmit and receieve */
+  kDifSpiHostSegmentTypeBidirectional,
+} dif_spi_host_segment_type_t;
+
+/**
+ * Address mode for the address segment in a transaction.
+ */
+typedef enum dif_spi_host_addr_mode {
+  /** The address is a 3-byte address. */
+  kDifSpiHostAddrMode3b,
+  /** The address is a 4-byte address. */
+  kDifSpiHostAddrMode4b,
+} dif_spi_host_addr_mode_t;
+
+/**
+ * Segment descriptor for each segment in a transaction.
```

Comment:
I would include a comment specifying that this is a tagged union.

Reply:
Done.